File: packages/engine/src/cpus/6502/instructions/tay.js
import { CycleType } from '..'

/**
 * Transfer Accumulator to Y
 *
 * Y = A
 *
 * @example
 * MODE          SYNTAX        HEX LEN TIM
 * Implied       TAY           $A8  1   2
 */
export function* tay() {
  this.y = this.acc
  this.zero = this.y === 0
  this.negative = (this.y & 0x80) > 0
  yield CycleType.GET
}

File: packages/engine/src/cpus/6502/instructions/tay.zig
const std = @import("std");
const CPU = @import("../cpu.zig").CPU;
const Addressing = @import("../opcode.zig").Addressing;

// Transfer Accumulator to Y
//
// Y = A
//
// MODE          SYNTAX        HEX LEN TIM
// Implied       TAY           $A8  1   2

pub fn tay(self: *CPU, addressing: Addressing) void {
    _ = addressing;
    self.y = self.acc;
    self.z = self.y == 0;
    self.n = (self.y & 0x80) > 0;
}
